src/java/org/apache/cassandra/db/SinglePartitionReadCommand.java

```diff
@@ -1284,16 +1284,19 @@ protected void appendCQLWhereClause(CqlBuilder builder)
             }
         }
 
-        // We put the row filter first because the clustering index filter can end by "ORDER BY"
-        if (!rowFilter().isEmpty())
-            builder.append(" AND ").append(rowFilter().toCQLString());
-
-        // Pass RowFilter.none() to avoid the clustering filter trying to modify our row filter
-        String filterString = clusteringIndexFilter().toCQLString(metadata(), RowFilter.none());
+        // For queries with clustering columns in the rowFilter (e.g., ALLOW FILTERING or secondary index),
+        // remove those conditions to avoid duplication since they'll be handled by the clusteringIndexFilter
+        RowFilter adjustedFilter = rowFilter();
+        for (RowFilter.Expression expr : rowFilter().expressions())
```

Comment:
Can it be `adjustedFilter().expressions()`?

Reply:
We're iterating through the original `rowFilter()` expressions to decide what to remove and as we iterate, we're building up `adjustedFilter` by removing items from it, so we can't switch.